ray_on_golem/server/services/golem/golem.py

```diff
@@ -82,11 +78,14 @@ async def shutdown(self) -> None:
         """
         logger.info("Stopping GolemService...")
 
-        await self._payment_manager.terminate_agreements()
+        if self._payment_manager is None:
```

Comment:
It is not clear why `self._payment_manager is None` indicates `cluster was not started`

Reply:
Well, it comes from the flow. `create_cluster` is the only place where `PaymentManager` is created, but at shutdown, we need take into the account, that server can be stopped before calling first `create_cluster`.